components/style/values/specified/mod.rs

```diff
@@ -345,14 +357,15 @@ impl Time {
         from_calc: bool)
         -> Result<Time, ()>
     {
-        let seconds = match_ignore_ascii_case! { unit,
-            "s" => value,
-            "ms" => value / 1000.0,
+        let (seconds, unit) = match_ignore_ascii_case! { unit,
+            "s" => (value, TimeUnit::Second),
+            "ms" => (value / 1000.0, TimeUnit::Millisecond),
```

Comment:
I wonder if we should avoid dividing here and just do it on the `seconds` function (renaming the `seconds` member).

Not a big deal probably anyway.